### PR TITLE
add tests for --empty flag

### LINKED ISSUE #1029: [ADAP-1035] [Tests] Add tests for --empty flag
Add tests associated with https://github.com/dbt-labs/dbt-core/pull/8971

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves #1029
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/# N/A

### Problem

https://github.com/dbt-labs/dbt-core/pull/8971 introduced a new `--empty` flag that required implementation changes to the BaseRelation.

### Solution

Add tests in dbt-bigquery to ensure expected behaviour of the new `--empty` flag when running with dbt-bigquery.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # install latest changes in dbt-core
 # TODO: how to automate switching from develop to version branches?
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-core&subdirectory=core
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter
+git+https://github.com/dbt-labs/dbt-core.git@dry-run-flag#egg=dbt-core&subdirectory=core
+git+https://github.com/dbt-labs/dbt-core.git@dry-run-flag#egg=dbt-tests-adapter&subdirectory=tests/adapter
```
Comment (MichelleArk) [inline dev-requirements.txt:4]: will be reverted before merging

**Review by mikealfare — APPROVED** [review]
